test/org_parser/parser_test.cljc

```diff
@@ -1004,3 +1029,60 @@ is another section"))))))
                [:table-formula "$4=vmean($2..$3)"]]]
              (parse " |--+--|\n | x|x |\n |--+--|\n #+TBLFM: $4=vmean($2..$3)"))))
     ))
+
+
+(deftest clock
+  (let [parse #(parser/org % :start :clock)]
+    (testing "a simple clock line"
+      (is (= [:clock [:timestamp-inactive-range
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:26"]]
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:46"]]]
+              [:clock-duration [:clock-dur-hh "0"] [:clock-dur-mm "20"]]]
+             (parse "CLOCK: [2021-05-22 Sat 23:26]--[2021-05-22 Sat 23:46] =>  0:20"))))
+    (testing "a simple clock line"
+      (is (= [:clock [:timestamp-inactive-range [:ts-inner-span
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:26"]]
+               [:ts-time "23:46"]
+               [:ts-modifiers]]]
+              [:clock-duration [:clock-dur-hh "0"] [:clock-dur-mm "20"]]]
+             (parse " CLOCK: [2021-05-22 Sat 23:26-23:46] =>  0:20 "))))
+    (testing "do not parse corrupted headlines"
+      (is (insta/failure? (parse "CLOCK: [not a timestamp Sat 23:26] =>  0:20 "))))
+    ))
+
+
+(deftest diary-sexp
+  (let [parse #(parser/org % :start :diary-sexp)]
+    (testing "a simple diary sexp line"
+      (is (= [:diary-sexp "nr()<n)-h"]
+             (parse "%%(nr()<n)-h"))))
+    (testing "do not parse if not starting at begin of line"
+      (is (insta/failure? (parse "  %%(x)"))))
+    ))
+
+
+(deftest planning
+  (let [parse #(parser/org % :start :planning)]
+    (testing "a simple planning info"
+      (is (= [:planning [:planning-info
+                         [:planning-keyword [:planning-kw-scheduled]]
+                         [:timestamp [:timestamp-inactive [:ts-inner [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:26"]] [:ts-modifiers]]]]]]
+             (parse "SCHEDULED: [2021-05-22 Sat 23:26]"))))
```

Comment:
Does it make sense to have inactive timestamps as a planning item? It will not show up in the agenda like this.

I've quickly tried to look up if there's a flag to configure it like this and couldn't find it.

Reply:
Thanks for the feedback :) I tried to stick to the worg spec. It says, the planning info has to contain a timestamp and links to the timestamps. It doesn't say, that it must be an active one. In other places the spec distinguishes between active and inactive, e.g. at the `CLOCK:` entries.

I look at inactive timestamps in planning information as "temporary disabled".

Reply:
Parsing them is certainly not wrong in case somebody does it, so it's pretty cool that your implementation can already do it(;

However, I haven't seen it so far and wouldn't see a reason to do it this way. The [docs on schedules and deadline](https://orgmode.org/manual/Deadlines-and-Scheduling.html) also show examples with active timestamps - albeit without explicitly saying that this is the only option(;

Reply:
It's true that only active planning info must appear in the agenda. We have to consider that later. I personally like the option to disable planning info without deleting the planning info, e.g. a cancelled event.

We should discuss anyway, how strict the parser should be. We can talk about this next weekend :)